pkg/reconciler/messaging/channel/channel.go

```diff
@@ -53,77 +58,20 @@ type Reconciler struct {
-func (r *Reconciler) Reconcile(ctx context.Context, key string) error {
-	// Convert the namespace/name string into a distinct namespace and name
-	namespace, name, err := cache.SplitMetaNamespaceKey(key)
-	if err != nil {
-		logging.FromContext(ctx).Desugar().Error("Invalid resource key")
-		return nil
-	}
-
-	// Get the Channel resource with this namespace/name
-	original, err := r.channelLister.Channels(namespace).Get(name)
-	if apierrs.IsNotFound(err) {
-		// The resource may no longer exist, in which case we stop processing.
-		logging.FromContext(ctx).Desugar().Error("Channel in work queue no longer exists")
-		return nil
-	} else if err != nil {
-		return err
-	}
-
-	// Don't modify the informers copy
-	channel := original.DeepCopy()
-
-	// Reconcile this copy of the Channel and then write back any status
-	// updates regardless of whether the reconciliation errored out.
-	var reconcileErr = r.reconcile(ctx, channel)
-
-	// If no error is returned, mark the observed generation.
-	if reconcileErr == nil {
-		channel.Status.ObservedGeneration = channel.Generation
-	}
-
-	if equality.Semantic.DeepEqual(original.Status, channel.Status) {
-		// If we didn't change anything then don't call updateStatus.
-		// This is important because the copy we loaded from the informer's
-		// cache may be stale and we don't want to overwrite a prior update
-		// to status with this stale state.
-
-	} else if uErr := r.updateStatus(ctx, original, channel); uErr != nil {
-		logging.FromContext(ctx).Desugar().Warn("Failed to update Channel status", zap.Error(uErr))
-		r.Recorder.Eventf(channel, corev1.EventTypeWarning, "UpdateFailed",
-			"Failed to update status for Channel %q: %v", channel.Name, uErr)
-		return uErr
-	} else if reconcileErr == nil {
-		// There was a difference and updateStatus did not return an error.
-		r.Recorder.Eventf(channel, corev1.EventTypeNormal, "Updated", "Updated Channel %q", channel.Name)
-	}
-	if reconcileErr != nil {
-		r.Recorder.Event(channel, corev1.EventTypeWarning, "InternalError", reconcileErr.Error())
-	}
-	return reconcileErr
-}
+// Check that our Reconciler implements Interface.
+var _ channelreconciler.Interface = (*Reconciler)(nil)
 
-func (r *Reconciler) reconcile(ctx context.Context, channel *v1alpha1.Channel) error {
+func (r *Reconciler) ReconcileKind(ctx context.Context, channel *v1alpha1.Channel) pkgreconciler.Event {
 	ctx = logging.WithLogger(ctx, r.Logger.With(zap.Any("channel", channel)))
 
 	channel.Status.InitializeConditions()
-
-	if channel.DeletionTimestamp != nil {
-		return nil
-	}
+	channel.Status.ObservedGeneration = channel.Generation
 
 	// 1. Create the Topic.
 	topic, err := r.reconcileTopic(ctx, channel)
 	if err != nil {
 		channel.Status.MarkTopicFailed("TopicReconcileFailed", "Failed to reconcile Topic: %s", err.Error())
-		return err
+		return pkgreconciler.NewEvent(corev1.EventTypeWarning, reconciledTopicFailedReason, "Reconcile Topic failed with: %s", err.Error())
```

Comment:
nit: usually I just use `%v` for `err`.
But also wonder if you have any special reason of doing what're doing now.

Reply:
thanks, no specific reason why I'm using this. I'll make one final pass to fix all of these, once all PRs are in, to leave them all in a consistent state